llvm/test/CodeGen/RISCV/copyprop.ll

```diff
@@ -35,13 +33,12 @@ define void @copyprop_after_mbp(i32 %v, ptr %a, ptr %b, ptr %c, ptr %d) {
 ; PROP-NEXT:    sw a0, 0(a2)
 ; PROP-NEXT:    li a0, 1
 ; PROP-NEXT:    sw a0, 0(a1)
-; PROP-NEXT:    li a0, 12
-; PROP-NEXT:    sw a0, 0(a4)
-; PROP-NEXT:    ret
+; PROP-NEXT:    j .LBB0_3
```

Comment:
Is this undoing tail duplication?

Reply:
How does this end up happening? I thought this pass ran after tail duplication

Reply:
One of the things the branch folding pass does is merging common tails. Tail duplication explicitly creates common tails to reduce the number branch instructions that will be executed. If you run branch folding after tail duplication it will undo explicit tail duplication.